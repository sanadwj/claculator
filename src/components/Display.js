import React from 'react';
import PropTypes from 'prop-types';
import '../styles/App.css';

const Display = props => {
  const { total, operation, next } = props;
  return (
    <h1 className="display">
      {(!total && next) || total }
      {operation }
      {(total && next)}
    </h1>
  );
};

Display.propTypes = {
  total: PropTypes.string,
  operation: PropTypes.string,
  next: PropTypes.string,
};

Display.defaultProps = {
  total: String,
  operation: String,
  next: String,
};

export default Display;
