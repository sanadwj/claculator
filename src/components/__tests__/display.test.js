import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import Display from '../Display';

describe('Display', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Display />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Creates a component', () => {
    render(<Display total="9" />);
    const element = screen.getByText('9');
    expect(element).toBeInTheDocument();
  });
});
