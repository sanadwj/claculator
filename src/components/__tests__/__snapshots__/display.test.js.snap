// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders correctly 1`] = `
<h1
  className="display"
/>
`;
